main.tf

```diff
@@ -207,15 +207,9 @@ resource "aws_sqs_queue" "queue_with_kms_and_redrive_and_no_policy" {
   tags = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment), map("KubernetesCluster", var.environment))}"
 }
 
-resource "aws_iam_user" "sqs_iam_user" {
-  count = "${length(var.policy) != 0 && length(var.kms_alias) == 0 ? var.number_of_users : 0}"
-
-  name = "${var.sqs_iam_user}${var.number_of_users != 1 ? "-${count.index}" : "" }"
-  path = "/"
-}
 
 resource "aws_iam_user" "sqs_with_kms_iam_user" {
```

Comment:
I'd rename this to not mention KMS (e.g.`sqs_iam_user`) and change the names wherever else it's needed.
Or put back the user you're deleting and delete this one instead.